File: classTypes4/js/AReptilBase.js
class ReptilBase extends AnimalBase {
    constructor(nombreCientifico, pesoKg, tieneEscamas) {
        super(nombreCientifico, pesoKg);
        this.tieneEscamas = tieneEscamas;
    }
    
    // implementacion concreta del metodo termoregular (de la interfaz reptil)
    termoregular() {
        console.log("el reptil esta tomando el sol para elevar su temperatura corporal");
    }

    // metodo abstracto para un movimiento especifico
    tipoDeLocomocion() {
    }
}

File: classTypes4/go/AReptilBase.go
package main

import "fmt"

// ReptilBase simula la clase abstracta.
type ReptilBase struct {
	AnimalBase // composicion
	tieneEscamas bool
}

// NuevoReptilBase es el constructor
func NuevoReptilBase(nombreCientifico string, pesoKg float64, tieneEscamas bool) ReptilBase {
	return ReptilBase{
		AnimalBase:   NuevoAnimalBase(nombreCientifico, pesoKg),
		tieneEscamas: tieneEscamas,
	}
}

// Termoregular implementacion concreta del metodo de la interfaz Reptil
func (r ReptilBase) Termoregular() {
	fmt.Println("el reptil esta tomando el sol para elevar su temperatura corporal")
}

// TipoDeLocomocion es el metodo abstracto, debe ser implementado por structs concretos.